reference/test/solver/lower_trs.cpp

```diff
@@ -0,0 +1,154 @@
+              {{2, 0.0, 0.0}, {3.0, 1, 0.0}, {1.0, 2.0, 3}}, exec)),
+          b(gko::initialize<Mtx>({{2, 0.0, 0.0}}, exec)),
+          csr_mtx(gko::copy_and_convert_to<CsrMtx>(exec, mtx.get())),
+          lower_trs_factory(Solver::build().on(exec)),
+          lower_trs_solver(lower_trs_factory->generate(mtx, b))
+    {}
+
+    std::shared_ptr<const gko::Executor> exec;
+    std::shared_ptr<Mtx> mtx;
+    std::shared_ptr<Mtx> b;
+    std::shared_ptr<CsrMtx> csr_mtx;
+    std::unique_ptr<Solver::Factory> lower_trs_factory;
+    std::unique_ptr<Solver> lower_trs_solver;
+
+    static void assert_same_matrices(const Mtx *m1, const Mtx *m2)
+    {
+        ASSERT_EQ(m1->get_size()[0], m2->get_size()[0]);
+        ASSERT_EQ(m1->get_size()[1], m2->get_size()[1]);
+        for (gko::size_type i = 0; i < m1->get_size()[0]; ++i) {
+            for (gko::size_type j = 0; j < m2->get_size()[1]; ++j) {
+                EXPECT_EQ(m1->at(i, j), m2->at(i, j));
+            }
+        }
+    }
+};
+
+
+TEST_F(LowerTrs, LowerTrsFactoryCreatesCorrectSolver)
+{
+    ASSERT_EQ(lower_trs_solver->get_size(), gko::dim<2>(3, 3));
+    ASSERT_NE(lower_trs_solver->get_system_matrix(), nullptr);
+    ASSERT_NE(lower_trs_solver->get_rhs(), nullptr);
+    ASSERT_EQ(lower_trs_solver->get_system_matrix(), mtx);
+    ASSERT_EQ(lower_trs_solver->get_rhs(), b);
+}
+
+
+TEST_F(LowerTrs, CanBeCopied)
+{
+    auto copy = Solver::build().on(exec)->generate(Mtx::create(exec),
+                                                   Mtx::create(exec));
+
+    copy->copy_from(lend(lower_trs_solver));
+
+    ASSERT_EQ(copy->get_size(), gko::dim<2>(3, 3));
+    auto copy_mtx = copy.get()->get_system_matrix();
+    auto copy_b = copy.get()->get_rhs();
+    assert_same_matrices(static_cast<const Mtx *>(copy_mtx.get()), mtx.get());
+    assert_same_matrices(static_cast<const Mtx *>(copy_b.get()), b.get());
+}
+
+
+TEST_F(LowerTrs, CanBeMoved)
+{
+    auto copy =
+        lower_trs_factory->generate(Mtx::create(exec), Mtx::create(exec));
+
+    copy->copy_from(std::move(lower_trs_solver));
+
+    ASSERT_EQ(copy->get_size(), gko::dim<2>(3, 3));
+    auto copy_mtx = copy.get()->get_system_matrix();
+    auto copy_b = copy.get()->get_rhs();
+    assert_same_matrices(static_cast<const Mtx *>(copy_mtx.get()), mtx.get());
+    assert_same_matrices(static_cast<const Mtx *>(copy_b.get()), b.get());
+}
+
+
+TEST_F(LowerTrs, CanBeCloned)
+{
+    auto clone = lower_trs_solver->clone();
+
+    ASSERT_EQ(clone->get_size(), gko::dim<2>(3, 3));
+    auto clone_mtx = clone.get()->get_system_matrix();
+    auto clone_b = clone.get()->get_rhs();
+    assert_same_matrices(static_cast<const Mtx *>(clone_mtx.get()), mtx.get());
+    assert_same_matrices(static_cast<const Mtx *>(clone_b.get()), b.get());
+}
+
+
+TEST_F(LowerTrs, CanBeCleared)
+{
+    lower_trs_solver->clear();
+
+    ASSERT_EQ(lower_trs_solver->get_size(), gko::dim<2>(0, 0));
+    auto solver_mtx = lower_trs_solver.get()->get_system_matrix();
```

Comment:
This is actually not the solver matrix you are testing with this (`csr_system_matrix_` would be it).